eva-lib/src/main/java/uk/ac/ebi/eva/lib/repository/VariantStudySummaryRepositoryImpl.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2016-2017 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.eva.lib.repository;
+
+import org.springframework.data.mongodb.MongoDbFactory;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.data.mongodb.core.aggregation.Aggregation;
+import org.springframework.data.mongodb.core.aggregation.AggregationResults;
+import org.springframework.data.mongodb.core.aggregation.Fields;
+import org.springframework.data.mongodb.core.aggregation.GroupOperation;
+import org.springframework.data.mongodb.core.aggregation.MatchOperation;
+import org.springframework.data.mongodb.core.aggregation.ProjectionOperation;
+import org.springframework.data.mongodb.core.aggregation.TypedAggregation;
+import org.springframework.data.mongodb.core.convert.MappingMongoConverter;
+import org.springframework.data.mongodb.core.mapreduce.GroupBy;
+import org.springframework.data.mongodb.core.mapreduce.GroupByResults;
+import org.springframework.data.mongodb.core.query.Criteria;
+import org.springframework.data.mongodb.repository.MongoRepository;
+
+import uk.ac.ebi.eva.commons.models.data.VariantSourceEntity;
+import uk.ac.ebi.eva.lib.repository.projections.VariantStudySummary;
+
+import java.util.List;
+
+import static org.springframework.data.mongodb.core.aggregation.Aggregation.group;
+import static org.springframework.data.mongodb.core.aggregation.Aggregation.match;
+import static org.springframework.data.mongodb.core.aggregation.Aggregation.project;
+
+/**
+ * This interface documents how studies can be queried.
+ * <p>
+ * This interface is used by Spring to create the query methods for VariantSourceEntity.
+ * Spring creates the implementation automatically by looking at the method name.
+ */
+public class VariantStudySummaryRepositoryImpl implements VariantStudySummaryRepositoryCustom {
+
+    private static final String STUDY_ID = "studyId";
+
+    private static final String STUDY_NAME = "studyName";
+
+    private static final String FILES_COUNT = "filesCount";
+
+    private static final String ID = "_id.";
+
+    private MongoTemplate mongoTemplate;
+
+    public VariantStudySummaryRepositoryImpl(MongoDbFactory mongoDbFactory, MappingMongoConverter mappingMongoConverter) {
+        mongoTemplate = new MongoTemplate(mongoDbFactory, mappingMongoConverter);
+    }
+
+    /**
+     * the equivalent intended query is:
```

Comment:
I put "the intended" because if there were a mismatch between both queries, the one in the comment is the intended one. If there were a mismatch without knowing which one is the correct, it might lead to "improve" the good one to a wrong query.